src/components/pages/CurrentCity/CurrentCity.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Button } from '@material-ui/core';
+import Grid from '@material-ui/core/Grid';
+import { Link } from "react-router-dom";
+import WeatherIcon from '../../reusable/WeatherIcon/WeatherIcon';
+import EditLocationIcon from '@material-ui/icons/EditLocation';
+import ArrowRightAltIcon from '@material-ui/icons/ArrowRightAlt';
+import CurrentDate from '../../reusable/CurrentDate/CurrentDate';
+import CurrentTime from '../../reusable/CurrentTime/CurrentTime';
+import {
+    windWrapper,
+    editLocationButtonWrapper,
+    editLocationButton,
+    windText,
+    windIcon,
+    date,
+    time,
+    city,
+    temperature
+} from './CurrentCity.scss';
+
+class CurrentCity extends React.Component {
+    constructor(props) {
+        super(props);
+        this.cityData = this.props.cityData;
```

Comment:
the problem you may face with is that when props change - your component (CurrentCity) won't reflect those changes. Constructor executes only once per Component's lifecycle setting props data to **this**. Then you are using instance (this) data do reflect it on a view. You may never notice in your specific ad-hoc case: say, your CurrentCity component being rerendered from the scratch every time executing constructor each time. But this kind of approach is bug prone.

Instead , why not just render data from props directly ?